http/src/main/java/org/sapia/ubik/rmi/server/transport/http/JdkRmiClientConnection.java

```diff
@@ -159,26 +220,52 @@ public Object receive(long timeout) throws IOException,
       IoUtils.closeSilently(is);
     }
   }
-
-  /**
-   * @see org.sapia.ubik.net.Connection#send(java.lang.Object)
-   */
-  public void send(Object o) throws IOException, RemoteException {
-    send(o, null, null);
-  }
-
-  JdkRmiClientConnection setUp(HttpAddress addr) throws RemoteException {
-    closed = false;
-
-    if (url == null) {
-      try {
-        url = new URL(addr.toString());
-      } catch (MalformedURLException e) {
-        throw new RemoteException(addr.toString(), e);
+  
+  protected void doSend(Object o, VmId associated, String transportType) throws IOException, RemoteException {
+    try {
+      conn = (HttpURLConnection) url.openConnection();
+      conn.setDoInput(true);
+      conn.setDoOutput(true);
+      conn.setUseCaches(false);
+      conn.setConnectTimeout(connectTimeOut);
+      conn.setReadTimeout(readTimeout);
+      conn.setRequestMethod(POST_METHOD);
+  
+      ByteArrayOutputStream bos = new ByteArrayOutputStream(bufsz);
+      ObjectOutputStream mos = MarshalStreamFactory.createOutputStream(bos);
+  
+      if ((associated != null) && (transportType != null)) {
+        ((RmiObjectOutput) mos).setUp(associated, transportType);
       }
+  
+      Split split = serializationTime.start();
+      mos.writeObject(o);
+      mos.flush();
+      mos.close();
+      split.stop();
+  
+      split = sendTime.start();
+      byte[] data = bos.toByteArray();
+  
+      if (data.length > bufsz) {
+        bufsz = data.length;
+      }
+  
+      conn.setRequestProperty(CONTENT_LENGTH_HEADER, "" + data.length);
+  
+      OutputStream os = conn.getOutputStream();
+      os.write(data);
+      os.flush();
+      os.close();
```

Comment:
don't know the semantic of the ubik binary protocol over http, but wondering why we are not looking at the http response on the connection?

Reply:
What do you mean by looking at the response? HTTP status code? Currently we're not leveraging that: HTTP is used simply as a transport and if an exception occured on the server, the deserialized payload will be an Exception, which will bubbled up and thrown to the calling thread on the client side. Any errors besides that will be deserialization/network errors, and in this case they will be wrapped in RemoteException

Reply:
ok, that's the logic I didn't know